sesman/chansrv/audin.c

```diff
@@ -88,6 +88,27 @@ static struct xr_wave_format_ex **g_client_formats = NULL;
 
 static int g_current_format = 0; /* index in g_client_formats */
 
+/*****************************************************************************/
+
+/*
+ * This can be called from sound.c because it includes audin.h.
+ */
+
+const char *
+audin_wave_format_tag_to_str(int tag)
+{
+    return
+        (tag == WAVE_FORMAT_PCM)        ? "WAVE_FORMAT_PCM" :
+        (tag == WAVE_FORMAT_ADPCM)      ? "WAVE_FORMAT_ADPCM" :
+        (tag == WAVE_FORMAT_ALAW)       ? "WAVE_FORMAT_ALAW" :
+        (tag == WAVE_FORMAT_MULAW)      ? "WAVE_FORMAT_MULAW" :
+        (tag == WAVE_FORMAT_MULAW)      ? "WAVE_FORMAT_MULAW" :
+        (tag == WAVE_FORMAT_MPEGLAYER3) ? "WAVE_FORMAT_MPEGLAYER3" :
+        (tag == WAVE_FORMAT_OPUS)       ? "WAVE_FORMAT_OPUS" :
+        (tag == WAVE_FORMAT_AAC)        ? "WAVE_FORMAT_AAC" :
+        NULL;
```

Comment:
Suggest something like "unknown" rather than NULL, as the returned vale from this function is passed straight into `LOG(LOG_LEVEL_INFO, "      wFormatTag      %s",` below.

Reply:
Indeed.

Reply:
That looks fine to me